apps/web/src/config/paymaster.ts

```diff
@@ -0,0 +1,27 @@
+import { zksyncTokens } from '@pancakeswap/tokens'
+
+import { ChainId } from '@pancakeswap/chains'
+import { Native, Currency, ERC20Token } from '@pancakeswap/sdk'
+import { Address } from 'viem'
+
+export const DEFAULT_PAYMASTER_TOKEN: PaymasterToken = {
+  ...Native.onChain(ChainId.ZKSYNC),
+  address: '0x0', // Only for compatibility
+}
+
+export const paymasterTokens: PaymasterToken[] = [
+  zksyncTokens.wbtc,
+  zksyncTokens.dai,
+  zksyncTokens.usdc,
+  zksyncTokens.usdcNative,
+  zksyncTokens.usdt,
+  zksyncTokens.grai,
+  zksyncTokens.tes,
+].map((token) => ({ ...token, discount: '-20%' }))
+
+export interface BasePaymasterToken {
+  address: Address
+  discount?: string
+}
+
+export type PaymasterToken = BasePaymasterToken & Partial<Currency | ERC20Token>
```

Comment:
How about
```suggestion
export type PaymasterToken = Currency & {
  discount?: string
}
```

Reply:
Causes type issues everywhere. 

The core of the issue I think is that all tokens are ERC20 except the Native one, and the types differ in that way. Like native doesn't have address and isNative & isToken are fixed boolean fields for both `ERC20Token` and `Native`.

Maybe I can use `null` value to mean Native. A getter for feeToken could return Native if null.

Reply:
For native token you can always `wrapped` to get the wrapped Token instance